### ISSUE TITLE
Unable to use V10 blob and queue SDKs in the same project

### ISSUE THREAD

**Issue by jasonbrome** [Azure/azure-storage-java#416]
### Which service(blob, file, queue, table) does this issue concern?
Blob, Queue

### Which version of the SDK was used?
azure-storage-blob 10.3.0
azure-storage-queue 10.0.0-Preview

### What problem was encountered?
Both com.microsoft.azure.storage.queue.StorageURL and com.microsoft.azure.storage.blob.StorageURL both reference com.microsoft.azure.storage.GeneratedStorageClient.  The version of GeneratedStorageClient is each library varies considerably from one version to the other - the first indication of this was a NoSuchMethodError when attempting to invoke dequeue on a MessageURL instance:

```
java.lang.NoSuchMethodError: com.microsoft.azure.storage.GeneratedStorageClient.generatedMessages()Lcom/microsoft/azure/storage/GeneratedMessages;
	at com.microsoft.azure.storage.queue.MessagesURL.dequeue(MessagesURL.java:199) ~[azure-storage-queue-10.0.0-Preview.jar:na]
```

Debugging the code showed the the GeneratedStorageClient from the Blob storage library was taking precendence, and that does not have the queue-related methods within it.

### Have you found a mitigation/solution?
Not yet - is the plan to consolidate on a single GeneratedStorageClient used across both projects (would they be harmonized on version?), or would each contain its own unique GeneratedStorageClient in a different package?

Thanks.

**Comment by rickle-msft** [comment]
Hi, @jasonbrome. Thank you for reporting this. Off the top of my head, they'll probably stay separate types because we want the packages to be completely independent, so we'll have to do some thinking on our end about how we want to get rid of the overloaded name. We might do something like the blob generated files are prefixed with GeneratedB and queue generated files are prefixed with GeneratedQ, which would disambiguate but isn't very pretty. 

I'll keep you posted on how this discussion goes around our team.

**Comment by rickle-msft** [comment]
We could! Organizationally, it's nice to keep them out of the blob/queue packages because blob and queue are where we do all of our actual work and handwrite code. The other packages are auto-generated and generated files can become noisy and pollute the workspace. That said, they are all prefixed with "Generated", so it shouldn't be too much of an issue. Thanks for the suggestion!

**Comment by LiamMorrow** [comment]
Any news on this one?  This is a major blocker for us

**Comment by rickle-msft** [comment]
A breaking release for blobs is still waiting on the new service version, so I can't do much there (changing the names of the types or their package would be a break). Now that I think about it, though, queues is still in preview and our versioning policy allows breaking in preview pretty freely, so I can make a change in queues that disambiguates the type names to at least mitigate the issue for now if that would help?

**Comment by LiamMorrow** [comment]
That would be ideal.  We were considering using

**Comment by LiamMorrow** [comment]
That would be awesome thank you.  We were considering using v8 otherwise, but it's such a big change between them.  Also it was unclear whether v8 supported Managed Identities for auth

**Comment by rickle-msft** [comment]
Alright. I'm thinking I can get to a release of blobs (for some other fixes) and queues (for this fix) likely by the end of tomorrow. We are definitely happy to facilitate your onboarding to v10 as that is our path forward, especially in the long term.

**Comment by rickle-msft** [comment]
Update: I have everything prepared and staged for a release. There is an unexpected issue in our signing pipeline that we are working on resolving. Thank you guys for your patience and sorry for the inconvenience.

**Comment by LiamMorrow** [comment]
Thank you for all your help @rickle-msft , will do

**Comment by rickle-msft** [comment]
We have released v11.0.0 of blobs which resolves this issue from that angle. I will close this issue as I believe it has been resolved, but please feel free to reopen it if you continue to experience problems.
